tools/export/cces/__init__.py

```diff
@@ -0,0 +1,418 @@
+        return cls.TOOLCHAIN in target.supported_toolchains
+
+    @property
+    def flags(self):
+        """Returns a dictionary of toolchain flags.
+        Keys of the dictionary are:
+        cxx_flags    - c++ flags
+        c_flags      - c flags
+        ld_flags     - linker flags
+        asm_flags    - assembler flags
+        common_flags - common options
+
+        Skip macros because headless tools handles them separately
+        """
+        config_header = self.toolchain.get_config_header()
+        flags = {key + "_flags": copy.deepcopy(value) for key, value
+                    in self.toolchain.flags.iteritems()}
+        if config_header:
+            config_header = os.path.relpath(config_header,
+                                self.resources.file_basepath[config_header])
+            config_header = "\\\"" + self.format_path(config_header) + "\\\""
+            header_options = self.toolchain.get_config_option(config_header)
+            flags['c_flags'] += header_options
+            flags['cxx_flags'] += header_options
+        return flags
+
+    @staticmethod
+    def format_path(path):
+        """
+        Formats the given path relative to the project directory
+        """
+        return os.path.abspath(path).replace("\\", "\\\\")
+
+    @staticmethod
+    def clean_flags(container, flags):
+        """
+        Some flags are handled by CCES already, so there's no need
+        to include them twice.
+        """
+        for flag in container:
+            if flag in flags:
+                flags.remove(flag)
+
+    @staticmethod
+    def convert_common_options(prefix, options, flags):
+        """
+        Converts the common flags into CCES options and removes them
+        from the flags list
+        """
+        # remove these flags without converting to option
+        # since they are added by CCES
+        remove = [ "-c" ]
+        for flag in remove:
+            if flag in flags:
+                flags.remove(flag)
+
+        value = prefix + "option.dwarfversion.enumerated.v2"
+        for flag in flags:
+            if flag.startswith("-gdwarf"):
+                value = prefix + "option.dwarfversion.enumerated.v" + flag[-1]
+                flags.remove(flag)
+                break
+        option = Option("baseId", value)
+        options[prefix + "option.dwarfversion"] = option
+
+    @staticmethod
+    def convert_assembler_options(flags):
+        """
+        Converts the assembler flags into CCES options and removes them
+        from the flags list
+        """
+        options = {}
+
+        # remove these flags without converting to option
+        # since they are added by CCES
+        remove = [ "-x", "assembler-with-cpp" ]
+        for flag in remove:
+            if flag in flags:
+                flags.remove(flag)
+
+        booleans = {"-v": "arm.assembler.option.verbose",
+                    "-g": "arm.assembler.option.debuginfo"}
+
+        for flag in booleans:
```

Comment:
Since you use both the key and the value from the `booleans` variable, you can iterate through the key, value pairs instead of doing double lookups?
```python
for flag, setting in booleans.items():
    value = "false"
    if flag in flags:
        value = "true"
        flags.remove(flag)
    options[setting] = Option("baseId", value)